### PR TITLE
Update and query animals table

### PR DESCRIPTION
**Pull request by mumo-esther**
Add a column species of type string to your animals table. 
### Insert the following data:

1. Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
2. Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
3. Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
4. Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
5. Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
6. Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
7. Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.

**Inside a transaction** 

- [ ] update the animals table by setting the species column to unspecified.
- [ ]  Verify that change was made. 
- [ ] Then roll back the change and verify that the species columns went back to the state before the transaction.
unspecified

**Inside a transaction:**

- [ ] Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- [ ] Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- [ ] Commit the transaction.
- [ ] Verify that change was made and persists after commit.
- [ ] Take a screenshot of the results of your actions.

**Inside a transaction** 

- [ ] delete all records in the animals table, then roll back the transaction.
- [ ] After the rollback verify if all records in the animals table still exists. 
- [ ] Take a screenshot of the results of your actions
delete

.
**Inside a transaction:**

- [ ] Delete all animals born after Jan 1st, 2022.
- [ ] Create a savepoint for the transaction.
- [ ] Update all animals' weight to be their weight multiplied by -1.
- [ ] Rollback to the savepoint
- [ ] Update all animals' weights that are negative to be their weight multiplied by -1.
- [ ] Commit transaction
- [ ] Take a screenshot of the results of your actions.
savepoint

**Write queries to answer the following questions:**

- How many animals are there?
count

- How many animals have never tried to escape?
esc0

- What is the average weight of animals?
avgweight

- Who escapes the most, neutered or not neutered animals?
avgneuterd

- What is the minimum and maximum weight of each type of animal?
minmaxheight

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
lastquery

### CODE REVIEW

File: data.sql
Comment (KDlamini) [inline data.sql:1]: In your README file, kindly update the MIT license link and make sure that you add a working URL to your license. In order to keep your README professional, there should not be any broken links.

Please remove this duplicated project description sentence to clean up your documentation.

image

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @mumo-esther 🙋‍♀️,  

Great work so far! 💯 
You have done a great job querying and updating the animal's table. 🎊 🥳 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  Good PR title and description
✔️  Code is clean and organized in appropriate files
✔️  All SQL commands schema, data, and queries are added correctly

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AlphaNtihinduka — CHANGES_REQUESTED** [review]
## Hi @mumo-esther 🙋‍♂️
Good job working on implementing the changes required by the previous reviewer. You have done the design to do the expectation and incredibly well. 👏  

TO HIGHLIGHT:
- Your project is professional ✅
- Professional README file :100: 
- Well-structured files ✅

# STATUS: Required Changes :recycle: 

close

## Required changes :recycle: 
- [ ] As the previous reviewer suggested, kindly update the MIT license link and make sure that you add a working URL to your license. In order to keep your README professional, there should not be any broken links. The MIT license gives users :person_curly_hair:  express permission to reuse code for any purpose.
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or slack channel if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK :crossed_fingers: :four_leaf_clover:

**Review by alexander16108 — APPROVED** [review]
Hi @mumo-esther , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Your project is professional ✅
Professional README file 💯
Well-structured files ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tugsem — COMMENTED** [review]
## Invalid Code Review Request ❌ 

Hi @mumo-esther , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

<h2>You have submitted a project, that belongs to the previous milestone and has already been approved.</h2>

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**